Design-Process.md

```diff
@@ -0,0 +1,194 @@
+* While it is possible that some rejected features end up getting reconsidered, this is generally the final state for language features that are explicitly considered and
+  rejected during LDM
+
+## Language Design Team processes
+
+These are various processes and actions taken by LDT members during the development of a feature. Community members should not perform these actions unless invited to do so
+by an LDT member.
+
+### Steps to move a [Discussion](#proposed-feature) to a [Champion feature](#championed-feature)
+
+When an LDT member decides to champion a discussion, they take the following steps:
+
+1. Create a new proposal champion issue.
+    * If preferred, the LDT member can ask the original proposer to create this issue.
+    * Note: it can be easier to create the PR for step 6 first, get that into a ready-to-merge state, and then create the champion issue at that point, depending on the complexity
+      of the feature.
+2. Assign themselves to the champion issue.
+3. Apply the `Proposal Champion` label to the new issue, as well as to the original discussion.
+4. Link to the original discussion from the champion issue.
+5. Lock the proposal champion issue for comments to ensure that discussion continues in the discussion area, rather than on the champion issue.
+6. Fill out and check in a [proposal template](proposals/proposal-template.md) for the feature. Exact specese is not required, but there should be enough detail to have a
+   meaningful triage session.
+    * This is also something the LDT member can ask a community member to open a PR for, if they are willing.
+    * The filled out proposal should include a link to the champion issue for easy navigation.
+
+### Bringing open questions to LDM
+
+During the course of development of a feature, there are several different types of questions that need to get brought to LDM for answers. The most important overriding factor
+for any question is that there is a checked-in commit that contains the question. The document and commit will be linked as part of the notes so that future readers of the notes
+can understand the full context in which the question was asked.
+
+#### Alternative proposals, supplemental documentation
+
+As part of the initial design of a feature, a number of different proposals may be brought as part of the design process, either as alternatives to an initial design, or as
+supplemental materials to an existing design to help drive conversation in LDM. We want to keep these "supplemental" materials in one place, rather than scattered throughout
+the repo as different issues, discussions, and other documents. For such material, they should go in the [working group folder](meetings/working-groups/) for that feature. Not
+all features will have such a folder; indeed, most will not. For these documents, please check them in _before_ bringing them to an LDM. The LDM organizer should be able to
+link to an exact document, not to a PR.
```

Comment:
Nit: even with unmerged PRs, you can link to the exact document. So I'm not sure if requiring the PR to be merged is necessary just because "the LDM organizer should be able to link to an exact document".

Reply:
Think @333fred main concern is that he wants the LDM notes to link to the document version at the point the LDM discussion happened. Basically, instead of linking to paths in the repo, he wants to link to the SHA commit of the document.

 I agree that you can do this if you understand how the commit's work under the hood but not sure it's first class enough for what Fred wants here.

Reply:
My concern is that I want to convey the exact context in which a question was asked. PRs can change over time, and the context (what comments are on the pr, what the text of the pr description was) is not linkable. I do not want to try and wrangle that context. Further, it can muddy the water on what the current status is if some PR exists for a long period of time and just keeps updating, rather than being merged. So I want to keep things to be fully checked in as much as possible.

Reply:
What's challenging for me about the PR policy is:
- "Offline" feedback on proposals is often best done through PR comments.
- But if the PR is already merged, then leaving the comments can feel like "drive-by"--the author is done with this PR, and now you're showing up and asking them to change it? How, by opening another PR?
- The process of checking if the comments have been addressed in any subsequent PR(s) is also complicated by this.
- The policy of "the PR must be merged prior to LDM" tends to encourage author to merge sooner rather than later, in order to ensure the step is not forgotten prior to meeting, which means we are more likely to be in the above situation.

I imagine the PR author necessarily wants things to reach a stable state prior to LDM, so that they have a strong sense of what they are going to be presenting. I don't think we want a situation of merging the PR 1 hour before meeting, and having feedback and updates going in immediately before and so on, and being unsure what exactly the latest state of things was.

So, perhaps it would be reasonable to establish a *guideline*, that when someone opens a PR for an LDM topic, that the PR remains open for as long as author is interested in receiving "offline feedback" on it. For example, until 1-2 days in advance of discussing the topic, or as far in advance as the author needs depending on their time constraints. There are no sign-off requirements.

If we hold to the rule that PRs must be merged prior to LDM discussion, then it seems reasonable for the author to add a topic to the schedule by linking to a PR while it is still open, on the assumption that after the meeting, we change the link to the *merge commit* of the PR--representing the single, permanent, immutable record of what was discussed in that particular meeting. In that case, nothing in the PR description, comments, etc. or any other ephemeral artifacts should be considered part of *what the presenter brought for discussion*.

Reply:
I don't have a particular issue with that approach @RikkiGibson. My one concern with it is that I do not want to need to nag people to merge their PRs, which I do today. That's why I put the specific wording around the organizer being able to link a specific document.